### PR TITLE
ELPP-2908 Change for parameter to be AND rather than OR

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Also adds fuzzy matching.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Not sure if these need to be applied the boostings.

File: tests/src/Search/Web/ForTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            $this->getArticleFixture(3),
+        ]);
+
+        $this->newClient();
+        $this->jsonRequest('GET', '/search', ['for' => 'Robert Smiht']);
+        $response = $this->getJsonResponse();
+        $this->assertEquals($response->total, 2);
+        $this->assertEquals($response->items[0]->id, '15278');
+        $this->assertEquals($response->items[1]->id, '15276');
+    }
```
Comment (giorgiosironi) [inline tests/src/Search/Web/ForTest.php]: would add `Robert $inventedWord` and check `total` is `0`

Comment (thewilkybarkid) [inline tests/src/Search/Web/ForTest.php]: 👍

**Comment by thewilkybarkid** [comment]
Looking at it, we might want to use [simple_query_string](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-simple-query-string-query.html), as that allows for phrases (eg `foo "bar baz"`).

**Comment by thewilkybarkid** [comment]
Hmm, not sure how that works with boosting. Will leave for now.

**Pull request merged**